tracer/src/Datadog.Trace/Propagators/SpanContextPropagator.cs

```diff
@@ -165,9 +165,15 @@ public void Inject<TCarrier>(SpanContext context, TCarrier carrier, Action<TCarr
 
                     if (localSpanContext is not null && spanContext is not null)
                     {
-                        if (localSpanContext.RawTraceId == spanContext.RawTraceId)
+                        if (localSpanContext.RawTraceId == spanContext.RawTraceId || localSpanContext.TraceId128.Lower == spanContext.TraceId128.Lower)
```

Comment:
slight nitpick, but I think it'd be more consistent to swap the `RawTraceId` equality checks to just use the overridden `==` operator for the `TraceId` - it'll check the `Lower` and `Upper` and won't do string equality.

```suggestion
                        if (localSpanContext.TraceId128== spanContext.TraceId128 || localSpanContext.TraceId128.Lower == spanContext.TraceId128.Lower)
```